src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/model/DomainEventSummary.kt

```diff
@@ -1,16 +1,19 @@
 package uk.gov.justice.digital.hmpps.approvedpremisesapi.model
 
 import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.DomainEventType
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.UserEntity
 import java.sql.Timestamp
+import java.time.OffsetDateTime
 import java.util.UUID
 
 interface DomainEventSummary {
   val id: String
   val type: DomainEventType
-  val occurredAt: Timestamp
+  val occurredAt: OffsetDateTime
```

Comment:
This looks like a big benefit to using JPA QL rather than raw SQL. Should we be aiming to re-write out SQL functions generally into JPA QL? It would be good to understand whether this is one of our "tech debt" tasks to chip away at...

Reply:
I think it's something we need to decide on a case by case basis. If we're finding that our Native SQL is replicating what JPA can do for us (e.g. populating entity objects), we should probably use JPA QL. But - i find it harder to write personally and it can result in some cumbersome/heavy SQL if we're not careful. Definitely something to keep in mind!